apps/roam/src/utils/getAllDiscourseNodesSince.ts

```diff
@@ -1,48 +1,171 @@
-import getDiscourseNodes from "./getDiscourseNodes";
+import getDiscourseNodes, { DiscourseNode } from "./getDiscourseNodes";
 import findDiscourseNode from "./findDiscourseNode";
+import { OnloadArgs } from "roamjs-components/types";
+import getDiscourseNodeFormatExpression from "./getDiscourseNodeFormatExpression";
 
-type RoamDiscourseNodeData = {
+type ISODateString = string;
+
+export type RoamDiscourseNodeData = {
   author_local_id: string;
+  author_name: string;
   source_local_id: string;
   created: string;
+  vector: number[];
   last_modified: string;
-  author_name: string;
   text: string;
+  type: string;
+  node_title?: string;
 };
 
-type ISODateString = string;
+export type DiscourseNodesSinceResult = {
+  pageNodes: RoamDiscourseNodeData[];
+  blockNodes: RoamDiscourseNodeData[];
+};
+
+export const getDiscourseNodeTypeWithSettingsBlockNodes = (
+  node: DiscourseNode,
+  sinceMs: number,
+  extensionAPI: OnloadArgs["extensionAPI"],
```

Comment:
Will update this to use configTree when this is merged to main and rebase the UI branch off it